modules/user/lib/user-sync-gauge-balance.service.ts

```diff
@@ -112,32 +108,63 @@ export class UserSyncGaugeBalanceService implements UserStakedBalanceService {
             we need to figure out which users have a changed balance on any gauge contract and update their balance,
             therefore we check all transfer events since the last synced block
          */
-
-        const events: ethers.providers.Log[] = [];
-        const logPromises: Promise<ethers.providers.Log[]>[] = [];
         const erc20Interface = new ethers.utils.Interface(ERC20Abi);
 
-        console.log(`user-sync-staked-balances-${networkContext.chainId} getLogs of ${gaugeAddresses.length} gauges`);
-
-        for (const gaugeAddress of gaugeAddresses) {
-            logPromises.push(
-                networkContext.provider.getLogs({
-                    //ERC20 Transfer topic
-                    topics: ['0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef'],
-                    fromBlock: startBlock,
-                    toBlock: endBlock,
-                    address: gaugeAddress,
-                }),
-            );
-        }
-        const allResponses = await Promise.all(logPromises);
+        // Split the range into smaller chunks to avoid RPC limits, setting up to 50 times max block range
+        const maxBlockRange = networkContext.data.rpcMaxBlockRange;
+        const toBlock = Math.min(startBlock + 50 * maxBlockRange, latestBlock);
+        const range = toBlock - startBlock;
+        console.log(`user-sync-staked-balances-${networkContext.chainId} block range from ${startBlock} to ${toBlock}`);
```

Comment:
Looks good, but would also isolate the networkContext as you did in the wallet sync service

Reply:
done